MapboxNavigation/NavigationViewController.swift

```diff
@@ -368,10 +369,11 @@ public class NavigationViewController: NavigationPulleyViewController, RouteMapV
             }
         }
         
-        if destination == nil {
-            let annotation = MGLPointAnnotation()
-            annotation.coordinate = route.coordinates!.last!
-            destination = annotation
+        for leg in route.legs {
```

Comment:
If we’re only looking at the last step, there’s no need to iterate over all the steps:

```swift
if let lastStep = route.legs.last?.steps.last {
```